### DISCUSSION TITLE
Where can I find the pin-assignment for Polarfire SoC Icicle Kit at Buildroot Linux setting and the procedure to implement the "menuconfig and patch a new package to Polarfire SoC Icicle Kit

### DISCUSSION THREAD

**Discussion by chu551122** [polarfire-soc/polarfire-soc-documentation#164] [Linux build systems for PolarFire SoC] [answered]
Sorry to bother you, I want to learn more about the Pin-Assignment, I tried to search the device tree and the include directory but can’t find right file to define for Polarfire SoC Icicle Kit. By the way, for menuconfig, is it only “buildroot_initramfs_menuconfig” for Polarfire SoC Icicle Kit or other “menuconfig” can be used? What are the procedures to patch a new package to this Polarfire SoC Icicle Buildroot SDK? 

Thank you very much!

**Comment by ConchuOD** [comment, marked as answer]
a) For the pinout, have a look at the reference design: https://github.com/polarfire-soc/icicle-kit-reference-design

b) If run from the top level SDK directory, yes. But you can always change directory into the buildroot sub-directory and use whatever make commands you wish.

c) Any changes you make in the buildroot sub-directory will be reflected in the built image if you do a "make clean; make DEVKIT=\<foo\>"

**Reply by chu551122** [reply]
Sorry that I described not clearly, I didn’t want the H/W pin assignment for Polarfire SoC Icicle Kit.
I want to know the Pin define for Polarfire SoC in device tree.
Thank you very much!

**Reply by ConchuOD** [reply]
There is no pinctrl driver & therefore no defines for the dt are required.

**Reply by chu551122** [reply]
If I want to use Polarfire SoC for other Board Design instead of Icicle Kit, any recommendation for the files I need to modify for BSP from Polarfire SoC Icicle Kit Buildroot Linux SDK base?
Thank you very much!

**Reply by chu551122** [reply]
I'm not sure if using the below asking is correct : for Polarfire SoC Icicle Kit in "bare-metal" application, we have to add the "icicle_MSS_cfg.xml" to define the pins related to MSS; but for Yocto or Buildroot, which file and where it located to define the same purpose?
Thank you very much!

**Reply by ConchuOD** [reply]
> I'm not sure if using the below asking is correct : for Polarfire SoC Icicle Kit in "bare-metal" application, we have to add the "icicle_MSS_cfg.xml" to define the pins related to MSS; but for Yocto or Buildroot, which file and where it located to define the same purpose? Thank you very much!

The xml is not used by the BSP/SDK. The xml is provided to the HSS instead:
https://github.com/polarfire-soc/hart-software-services/blob/master/boards/mpfs-icicle-kit-es/soc_fpga_design/xml/ICICLE_MSS_mss_cfg.xml

> If I want to use Polarfire SoC for other Board Design instead of Icicle Kit, any recommendation for the files I need to modify for BSP from Polarfire SoC Icicle Kit Buildroot Linux SDK base? Thank you very much!

In the top level of the sdk run `find . -name "*m100pfsevp*"` and you'll see all the custom files needed for adding that board. Mainly you're looking at a new subdirectory in conf/foo and a new devicetree in linux/arch/riscv/boot/microchip/vendor,mpfs-foo.dts. Similarly, have a look at the board directory in the HSS, because you'll likely need one of those too:
https://github.com/polarfire-soc/hart-software-services/tree/master/boards/aries-m100pfsevp

**Reply by chu551122** [reply]
I'll study your link for reference. Thank you very much!

**Comment by chu551122** [comment]
Sorry that I described not clearly, I didn’t want the H/W pin assignment for Polarfire SoC Icicle Kit.
I want to know the Pin define for Polarfire SoC in device tree.
Thank you very much!

***@***.***
Best Regards

Michael Chu

******* Microchip Technology Inc. *******
Embedded Solutions Engineering Manager
Office: 886-2-25088633 (Direct)
Mobile: 886-925-717-311
WEB:  www.microchip.com<http://www.microchip.com/>
         www.microchip.com.tw<http://www.microchip.com.tw/>

Welcome to Register Microchip University
https://secure.microchip.com/mu

***@***.***
